lib/plugins/aws/deployFunction.js

```diff
@@ -90,8 +121,25 @@ class AwsDeployFunction {
     return data.Arn;
   }
 
+  // TODO: TEST
   async callUpdateFunctionConfiguration(params) {
-    await this.provider.request('Lambda', 'updateFunctionConfiguration', params);
+    // TODO: ADD max count?
+    const callWithRetryUntilSuccess = async () => {
+      try {
+        await this.provider.request('Lambda', 'updateFunctionConfiguration', params);
+      } catch (err) {
+        if (err.providerError && err.providerError.code === 'ResourceConflictException') {
+          this.serverless.cli.log(
+            `Retrying update of function: ${this.options.function}. Reason: ${err.message}`
+          );
+          await wait(5000);
```

Comment:
I would do `1s`. I think there's no harm with issuing request more frequently, and we can ensure that way use won't wait too long (5s adds significant overhead)

Reply:
I was even considering going higher here, the problem is that it takes ~25-30 seconds in my case to actually succeed with an update and I can imagine it might be even longer for specific use cases. Do you think it's a good approach to issue 25-30 retries under regular circumstances?

Reply:
I guess it's not that harmful. Context is most likely that some developer is updating the dev stage, and changed both image uri and some config (but that's to be confirmed I believe). So case is quite rare and should not affect production deployments.

Additionally we have some backoff logic in case of rate limit errors. So it's not that we'll just crash in such case.

Reply:
Yeah, in general, that case should be quite rare if we have proper checking for configuration update only in case when something really changed.